tests/unit_tests/mlocker.cpp

```diff
@@ -46,14 +46,14 @@ TEST(mlocker, distinct_1)
   const size_t base_pages = epee::mlocker::get_num_locked_pages();
   const size_t base_objects = epee::mlocker::get_num_locked_objects();
   std::unique_ptr<char[]> data{new char[8 * page_size]};
-  epee::mlocker *m0 = new epee::mlocker(BASE(data), 1);
-  epee::mlocker *m1 = new epee::mlocker(BASE(data) + 2 * page_size, 1);
-  epee::mlocker *m2 = new epee::mlocker(BASE(data) + 3 * page_size, 1);
+  std::shared_ptr<epee::mlocker> m0{new epee::mlocker(BASE(data), 1)};
+  std::shared_ptr<epee::mlocker> m1{new epee::mlocker(BASE(data) + 2 * page_size, 1)};
+  std::shared_ptr<epee::mlocker> m2{new epee::mlocker(BASE(data) + 3 * page_size, 1)};
   ASSERT_EQ(epee::mlocker::get_num_locked_pages(), base_pages + 3);
   ASSERT_EQ(epee::mlocker::get_num_locked_objects(), base_objects + 3);
-  delete m0;
-  delete m1;
-  delete m2;
+  m0 = NULL;
```

Comment:
Maybe `m0.reset(nullptr);`?

Reply:
It's just how I'm used to seeing smart-pointers reset, think it has same effect.